src/Datadog.Trace/Vendors/StatsdClient/StatsdUDP.cs

```diff
@@ -6,15 +6,17 @@
 using System.Net;
 using System.Net.Sockets;
 using System.Text;
-using System.Collections.Generic;
 
 namespace Datadog.Trace.Vendors.StatsdClient
 {
     internal class StatsdUDP : IDisposable, IStatsdUDP
     {
+        private static readonly object EndpointGate = new object();
+        private static int? _portToForce;
+        private static IPEndPoint _ipEndpoint;
+
         private int MaxUDPPacketSize { get; set; } // In bytes; default is MetricsConfig.DefaultStatsdMaxUDPPacketSize.
         // Set to zero for no limit.
-        public IPEndPoint IPEndpoint { get; private set; }
         private Socket UDPSocket { get; set; }
         private string Name { get; set; }
         private int Port { get; set; }
```

Comment:
You should be able to get rid of this property if you're going to handle everything statically.

Reply:
Didn't I already?